### ISSUE TITLE
TPM2: DecodeSignature fails to decode CertifyCreation signature by an ECC key

### ISSUE THREAD

**Issue by ElMostafaIdrassi** [google/go-tpm#191]
I've added a [test](https://github.com/ElMostafaIdrassi/go-tpm/commit/fdcd1b95dfa7b0de7f44c9fc6f53128c8666f3ee) for this particular case, as I found `TestCreateAndCertifyCreation` only deals with RSA keys.

As described in the commit, the test fails because the signature does not seem to be valid, mainly because of its small length (32 bytes during my tests).
It should normally consist of both S and R which both need to be, in this test case, 32-bytes long.

**This was tested using the simulator and a real TPM 2.0 device, under Linux.**

I'm providing the test case here as well : 

```
func TestCreateAndCertifyCreationECC(t *testing.T) {
	rw := openTPM(t)
	defer rw.Close()
	params := tpm2.Public{
		Type:       tpm2.AlgECC,
		NameAlg:    tpm2.AlgSHA256,
		Attributes: tpm2.FlagSignerDefault,
		ECCParameters: &tpm2.ECCParams{
			Sign: &tpm2.SigScheme{
				Alg:  tpm2.AlgECDSA,
				Hash: tpm2.AlgSHA256,
			},
			CurveID: tpm2.CurveNISTP256,
		},
	}
	keyHandle, pub, _, creationHash, tix, _, err := CreatePrimaryEx(rw, HandleEndorsement, pcrSelection7, emptyPassword, emptyPassword, params)
	if err != nil {
		t.Fatalf("CreatePrimaryEx failed: %s", err)
	}
	defer FlushContext(rw, keyHandle)

	scheme := SigScheme{Alg: AlgECDSA, Hash: AlgSHA256, Count: 0}
	attestation, signature, err := CertifyCreation(rw, emptyPassword, keyHandle, keyHandle, nil, creationHash, scheme, tix)
	if err != nil {
		t.Fatalf("CertifyCreation failed: %s", err)
	}
	att, err := DecodeAttestationData(attestation)
	if err != nil {
		t.Fatalf("DecodeAttestationData(%v) failed: %v", attestation, err)
	}
	if att.Type != TagAttestCreation {
		t.Errorf("Got att.Type = %v, want TagAttestCreation", att.Type)
	}
	p, err := DecodePublic(pub)
	if err != nil {
		t.Fatalf("DecodePublic failed: %v", err)
	}
	match, err := att.AttestedCreationInfo.Name.MatchesPublic(p)
	if err != nil {
		t.Fatalf("MatchesPublic failed: %v", err)
	}
	if !match {
		t.Error("Attested name does not match returned public key.")
		t.Logf("Name: %v", att.AttestedCreationInfo.Name)
		t.Logf("Public: %v", p)
	}

	sig, err := DecodeSignature(bytes.NewBuffer(signature))
	if err != nil {
		t.Fatalf("DecodeSignature failed: %v", err)
	}

	var pkEcdsa ecdsa.PublicKey
	var hsh hash.Hash
	pkEcdsa = ecdsa.PublicKey{Curve: elliptic.P256(), X: p.ECCParameters.Point.X(), Y: p.ECCParameters.Point.Y()}
	signHash, err := p.ECCParameters.Sign.Hash.Hash()
	if err != nil {
		t.Fatalf("Hash failed: %v", err)
	}
	hsh = signHash.New()
	hsh.Write(attestation)

	if !ecdsa.Verify(&pkEcdsa, hsh.Sum(nil), sig.ECC.R, sig.ECC.S) {
		t.Fatalf("Verify failed: %v", err)
	}
}
```

**Comment by ElMostafaIdrassi** [comment]
Hello @twitchy-jsonp 

I've managed to find what caused the signature to be truncated.
I've created 2 PRs : #192 which fixes the bug and #193 adds a test that validates the fix.

**Comment by ElMostafaIdrassi** [comment]
Closed by PR #192.
